### ISSUE TITLE
Recommended hardware?

### ISSUE THREAD

**Issue by hbadrane** [darrenabbey/ymap#14]
Hello, and thank you for this very nice tool.
What is the recommended hardware config so we can run let's say a 100 genomes in a reasonable time?

Best Regards,
Hassan

**Comment by darrenabbey** [comment]
My personal computer has an 8 core AMD 64 but CPU and 36 GB of RAM.

When running 7 datasets through YMAP, with 100% allocation of a CPU per
dataset, it can take a few hours to get through them all.

Running more datasets at once will slow down execution of each individual
dataset, but it still works (assuming no datasets have issues I hadn't
accounted for).

If I didn't have anything else to do, I could easily run through 20-30 in a
day in batches of 10 at a time, taking about a week to get through 100.

Setting up all the genomes to process at once would mean you wouldn't get
any results until everything is done, so I tend not to go this route.

Processing through datasets at a much faster rate would likely require a
computer cluster, but I have had no experience with setting up YMAP in that
environment.

- Darren

**Comment by darrenabbey** [comment]
Also, at this time, YMAP does not take advantage of GPUs for faster
processing of parallelizable tasks.

- Darren

**Comment by hbadrane** [comment]
Hey Darren, thank you so much for the inputs.
The reason we are looking into installing this locally, is that my colleague was using the UMN server but it looks like they have to take it down for some security reasons, so we can't access it anymore. Will stay in touch...

**Comment by darrenabbey** [comment]
I'm actively working with the Myer's lab at UMN to get it back online. I'm
expecting it'll be up in the next couple weeks.

I'd spend much of the last few months rewriting parts of YMAP to resolve
the security issues that had been found.

That said, I can still recommend running it on a local server as data
upload times will be reduced and there would be less security concerns in
the future.

- Darren

**Comment by hbadrane** [comment]
Yes, this is a so nice tool, that we should have a local copy running.
Do you think this can work on AWS ?

**Comment by darrenabbey** [comment]
Right now much of YMAP is Matlab based. I'm not sure Matlab will run in
that environment.

Eventually I plan to rewrite the Matlab portions into something that is
more accessible, like Python. That would probably be YMAP2. At that point
it would easily run on cloud services.

- Darren

**Comment by hbadrane** [comment]
I have no knowledge so far about AWS services, but why Matlab won't work?
So you mean it's best just to have a local lunix machine?

**Comment by darrenabbey** [comment]
I haven't played with AWS. I just know setting up Matlab requires a license
and that maybe be a complication in that environment.

- Darren

**Comment by hbadrane** [comment]
I see ... thanks.

**Comment by darrenabbey** [comment]
There is now a branch called "ymap_octave" that has Matlab stripped out, as well as a few other prereqs. This may be easier to run on AWS, if you're still looking at that option.
